### PR TITLE
[jsk_tools/test_topic_published.py] Add condition_%d option for checking topic value

### PR DESCRIPTION
**Pull request by iory**
# What is this?

Added condition option to check if the output topic value is correct.
The descriptions are followings.

```
- ``~condition_%d`` (``str``, Default: ``None``, Optional)

  Check bool value condition using the given Python expression.
  The Python expression can access any of the Python builtins plus:
  ``topic`` (the topic of the message), ``m`` (the message) and ``t`` (time of message).

  For example, topic is ``std_msgs/String`` and if you want to check whether a sentence is a ``hello``, you can do the following.

  .. code-block:: bash
    condition_0: m.data == 'hello'

  If you want to check the frame id of the header, you can do the following.

  .. code-block:: bash
    condition_0: m.header.frame_id in ['base', 'base_link']

  Note that, use escape sequence when using the following symbols ``<(&lt;)``, ``>(&gt;)``, ``&(&amp;)``, ``'(&apos;)`` and ``"(&quot;)``.

  .. code-block:: bash
    condition_0: m.data &lt; 'spbm'

```

https://github.com/jsk-ros-pkg/jsk_common/blob/cc9873caed9a27f2817e85000902c6244ea563a4/doc/jsk_tools/test_nodes/test_topic_published.rst

### CODE REVIEW

**Review by mqcmd196 — CHANGES_REQUESTED** [review]
If the topic published but not required the condition, the test says
 https://github.com/jsk-ros-pkg/jsk_common/pull/1744/files#diff-34c3faa1da3d534bbe400b78823970cc87424901f4008535bff83b3ee694cd32L117-L119, and it is confused a little. So could you make it says different things like `The topic %s not requires the condition` ?

File: doc/jsk_tools/test_nodes/test_topic_published.rst [hunk 1]
```diff
@@ -22,6 +22,22 @@ Parameters
...
+- ``~condition_%d`` (``str``, Default: ``None``, Optional)
+
+  Check bool value condition using the given Python expression.
+  The Python expression can access any of the Python builtins plus:
+  ``topic`` (the topic of the message), ``m`` (the message) and ``t`` (time of message).
+
+  For example, topic is ``std_msgs/String`` and if you want to check whether a sentence is a ``hello``, you can do the following.
+
+  .. code-block:: bash
+    condition_0: m.data == 'hello'
```
Comment (mqcmd196) [inline doc/jsk_tools/test_nodes/test_topic_published.rst:34]: Could you add the information about handling `>`, `<` in xml, like https://docs.oracle.com/cd/A97335_02/apps.102/bc4j/developing_bc_projects/obcCustomXml.htm ?

**Review by mqcmd196 — CHANGES_REQUESTED** [review]

**Comment by mqcmd196** [comment]
Thank you for your nice PR, it can make the test code much simpler

**Comment by mqcmd196** [comment]
I want to use in https://github.com/jsk-ros-pkg/jsk_recognition/pull/2695

File: doc/jsk_tools/test_nodes/test_topic_published.rst (hunk 1 above)
Comment (iory) [reply doc/jsk_tools/test_nodes/test_topic_published.rst:34]: Thanks for your comments.
I added the document. Please check it.
https://github.com/jsk-ros-pkg/jsk_common/blob/cc9873caed9a27f2817e85000902c6244ea563a4/doc/jsk_tools/test_nodes/test_topic_published.rst

**Comment by iory** [comment]
Thank you for your comments.
I updated the code to output a readable error message. Please check it.
https://github.com/jsk-ros-pkg/jsk_common/pull/1744/commits/5d4e2b98e7ad1147d3a504c95bd177becd080901

**Comment by iory** [comment]
Thanks! I fixed the lint error.

**Review by mqcmd196 — APPROVED** [review]

**Pull request merged**